apollo-router/src/plugins/rhai/engine.rs

```diff
@@ -218,6 +219,34 @@ mod router_method {
     }
 }
 
+#[export_module]
+mod status_code {
+    pub(crate) type StatusCode = http::StatusCode;
+
+    #[rhai_fn(name = "to_string", pure)]
+    pub(crate) fn status_code_to_string(status_code: &mut StatusCode) -> String {
+        status_code.as_str().to_string()
+    }
+
+    #[rhai_fn(name = "to_number", pure)]
+    pub(crate) fn status_code_to_number(status_code: &mut StatusCode) -> i64 {
+        status_code.as_u16() as i64
+    }
+
+    #[rhai_fn(name = "==", pure)]
+    pub(crate) fn status_code_equal_comparator(status_code: &mut StatusCode, other: i64) -> bool {
+        status_code.as_u16() as i64 == other
+    }
+
+    #[rhai_fn(name = "!=", pure)]
+    pub(crate) fn status_code_not_equal_comparator(
+        status_code: &mut StatusCode,
+        other: i64,
+    ) -> bool {
+        status_code.as_u16() as i64 != other
+    }
```

Comment:
I think I'd prefer comparators that were the same type, something like:
```
    #[rhai_fn(name = "==", pure)]
    pub(crate) fn status_code_equal_comparator(status_code: &mut StatusCode, other: &StatusCode) -> bool {
        status_code == other
    }
```
Same thing for `!=`

For this to make sense, we'd probably need a (fallible) constructor to make a StatusCode from a number.